src/lavinmq/queue/durable_queue.cr

```diff
@@ -211,46 +229,47 @@ module LavinMQ
               @ready = ready = ReadyQueue.new Math.pw2ceil(capacity)
             end
             next if acked.try { |a| a.bsearch { |asp| asp >= sp } == sp }
+            # SP refers to a segment that has already been deleted
+            unless @vhost.@segments.has_key? sp.segment
+              lost_msgs += 1
```

Comment:
are they lost? or are they messages that has already been acked? or both? potentially lost i guess..